core/src/main/java/org/pentaho/platform/util/ActionUtil.java

```diff
@@ -210,11 +212,53 @@ public static String generateWorkItemUid( final Map<String, Serializable> params
     // corresponding action keys (see KEY_MAP), fall back on the quartz key, if the action key cannot be found
     final String userName = Optional.ofNullable( (String) params.get( INVOKER_ACTIONUSER ) ).orElse( (String) params
       .get( QUARTZ_ACTIONUSER ) );
-    final String inputFilePath = Optional.ofNullable( (String) params.get( INVOKER_STREAMPROVIDER_INPUT_FILE ) )
+    String inputFilePath = Optional.ofNullable( (String) params.get( INVOKER_STREAMPROVIDER_INPUT_FILE ) )
       .orElse( (String) params.get( QUARTZ_STREAMPROVIDER_INPUT_FILE ) );
+
+    // if we still cannot find the inputFile, we have one last attempt: check for inline passing of input and output files,
+    // represented by 'ActionAdapterQuartzJob-StreamProvider -> input file = <path><colon>output file=<path>
+    if ( StringUtil.isEmpty( inputFilePath ) && isInlinePassingOfInputOnStreamProvider( params ) ) {
+      inputFilePath = getInlineInputFileOnStreamProvider( params );
+    }
+
     return generateWorkItemUid( inputFilePath, userName );
   }
 
+  public static boolean isInlinePassingOfInputOnStreamProvider( final Map<String, Serializable> params ) {
+
+    if ( params != null ) {
+
+      String streamProviderContent = ( params.containsKey( QUARTZ_STREAMPROVIDER )
+              ? params.get( QUARTZ_STREAMPROVIDER ) : params.containsKey( INVOKER_STREAMPROVIDER )
+              ? params.get( INVOKER_STREAMPROVIDER ) : StringUtils.EMPTY ).toString().trim();
+
+      return streamProviderContent.contains( QUARTZ_STREAMPROVIDER_INLINE_INPUT_FILE );
+    }
+
+    return false;
+  }
+
+  public static String getInlineInputFileOnStreamProvider( final Map<String, Serializable> params ) {
+
+    if ( isInlinePassingOfInputOnStreamProvider( params ) ) {
+
+      try {
+
+        String streamProviderContent = ( params.containsKey( QUARTZ_STREAMPROVIDER )
```

Comment:
minor refactoring: looks like same logic as line 231. Could be refactored out into a common method.

Reply:
Makes sense. If you agree, then I could tackle this in a follow-up "Cleanup" PR, rather than blocking this one from getting processed.